File: driver/languages/rust/template.rs
#![allow(warnings)]

// {{PART_MODULES}}

use parts::scanner::Scanner;
use parts::comparator::Comparator;
use parts::data_structures::{ListNode, TreeNode};

pub struct Solution;

{{USER_CODE}}

fn main() {
    let mut scanner = Scanner::new();
    
    let num_test_cases = match scanner.next_raw() {
        Some(s) => s.parse::<usize>().unwrap_or(0),
        None => 0,
    };

    for _ in 0..num_test_cases {
        let mut _phase = "parse_inputs";
        {{MAIN_LOOP}}
    }
}

File: driver/languages/go/template.go
package main

import (
	"bufio"
	"encoding/base64"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
// {{USER_IMPORTS}}
)

// {{PART_DATA_STRUCTURES}}

// {{PART_SCANNER}}

// {{PART_COMPARATOR}}

{{USER_CODE}}

func main() {
	scanner := NewScanner()
	comp := &Comparator{}
	_ = comp

	numTestCasesStr := scanner.NextRaw()
	numTestCases, err := strconv.Atoi(numTestCasesStr)
	if err != nil {
		return
	}

	for i := 0; i < numTestCases; i++ {
		_phase := "parse_inputs"
		func() {
			defer func() {
				if r := recover(); r != nil {
					fmt.Printf("@@ERROR@@:case=%d phase=%s msg=%v\n", i, _phase, r)
					os.Exit(1)
				}
			}()

			{{MAIN_LOOP}}
			_phase = "verify_sync"
			sentinel := scanner.NextRaw()
			if sentinel != "@@CASE_END@@" {
				panic(fmt.Sprintf("Input desync detected! Expected @@CASE_END@@ but got: %s", sentinel))
			}
		}()
	}
}
